iroh-bytes/src/util.rs

```diff
@@ -1,22 +1,158 @@
 //! Utility functions and types.
 use anyhow::Result;
 use bao_tree::blake3;
+use bytes::Bytes;
+use derive_more::{Debug, Display, From, Into};
 use postcard::experimental::max_size::MaxSize;
 use serde::{
     de::{self, SeqAccess},
     ser::SerializeTuple,
     Deserialize, Deserializer, Serialize, Serializer,
 };
-use std::{fmt, result, str::FromStr};
+use std::{borrow::Borrow, fmt, result, str::FromStr, time::SystemTime};
 use thiserror::Error;
 pub mod io;
 pub mod progress;
 pub mod runtime;
 
+/// A format identifier
+///
+/// Should we make this an u64 and use ?
+///
+/// That table is so weird. There is so much unrelated stuff in there, so the smallest value we would be
+/// able to use for iroh collections would be 2 bytes varint encoded or something...
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct BlobFormat(u64);
```

Comment:
I would strongly prefer to ignore multiformats entirely and use a `u8`, with `u8::MAX` being reserved for future extensions.

Reply:
Usually I agree. But in this case I am not so sure. This costs us almost nothing. We use 0x55 instead of 0 for "raw blob" and whatever other value we choose for collections instead of 1. Other than that this costs us nothing.

We are using postcard, so it's not like we are totally allergic to varints. And for cases where speed of serialization matters, BlobFormat serializes as just an u64.

In the doc db, we just got a 40 byte (hash, format) pair instead of a 33 byte pair, but then rounding up to word boundaries is probably a good idea anyway.

We could do some kind of permutation table thing where we replace 0x55 <-> 0 and the multiformats collection format id with 1, but what for?

And if we do an u8 that means we won't be able to ever represent all possible multiformats values, but have to have an explicit list of supported formats and mappings to our enum.